Sources/libhostmgr/S3Manager.swift

```diff
@@ -41,13 +46,23 @@ public struct S3Manager: S3ManagerProtocol {
     public func download(
         key: String,
         to destination: URL,
+        replacingExistingFile shouldReplaceExistingFile: Bool = true,
         progressCallback: FileTransferProgressCallback?
     ) async throws {
         let tempUrl = try await s3Client.download(
             objectWithKey: key,
             inBucket: self.bucket,
             progressCallback: progressCallback
         )
+
+        if FileManager.default.fileExists(at: destination) {
+            if shouldReplaceExistingFile {
+                try FileManager.default.removeItem(at: destination)
+            } else {
+                throw Errors.fileExistsAtDestination
+            }
+        }
+
         try FileManager.default.moveItem(at: tempUrl, to: destination)
```

Comment:
I presume the issue was caused by this `moveItem` call failure, and the fix is adding the option of replacing by default?

From [the API doc](https://developer.apple.com/documentation/foundation/filemanager/1414750-moveitem):
> If an item with the same name already exists at dstURL, this method stops the move attempt and returns an appropriate error.

Reply:
Yep that's correct